### PR TITLE
Refactor/upgrade to use setup cfg

### PR DESCRIPTION
**Pull request by misimpso**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,58 +1,43 @@
...
+	VENVPYTHON=$(PYTHONVENV)/Scripts/python
+else
+	VENVPYTHON=$(PYTHONVENV)/bin/python
+endif
+
+PHONY: init
+init:
+	@echo "Creating virtual environment 'venv'."
+	python3 -m venv venv || python -m venv venv
+	$(VENVPYTHON) -m pip install --upgrade setuptools pip
```
Comment (gsornsen) [inline Makefile:13]:
```suggestion
	$(VENVPYTHON) -m pip install --upgrade setuptools pip pip-tools
```

I'm suggesting adding `pip-tools` to this step so developers can generate  [requirements.txt](https://github.com/FuzzFoundation/WKTPlot/blob/refactor/upgrade-to-use-setup-cfg/requirements.txt#L1-L6)

File: requirements.txt [hunk 2]
```diff
@@ -1,3 +1,28 @@
...
+    # via bokeh
+packaging==21.3
+    # via bokeh
+pillow==8.4.0
+    # via bokeh
+pyparsing==3.0.6
+    # via packaging
+pyyaml==6.0
+    # via bokeh
+shapely==1.8.0
```
Comment (gsornsen) [inline requirements.txt:23]: This version doesn't exist on Mac, but you're not supporting Mac at the moment, and there are also issues. I may take a stab at Mac support in a separate PR if you don't mind.

File: setup.cfg [hunk 3]
```diff
@@ -0,0 +1,54 @@
+[metadata]
+name = wktplot
+version = attr: wktplot.__version__
```
Comment (gsornsen) [inline setup.cfg:3]: Are we keeping manual bump versioning for now?

File: setup.cfg [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+author = Michael Simpson, Gerald Sornsen
+author_email = mikeysimpson4@gmail.com, gerald@sornsen.io
+description = Well-known-text and shapely object visualiation library
+long_description = file: README.md
+long_description_content_type = text/markdown
+url = https://github.com/FuzzFoundation/WKTPlot
+platform = any
+license = MIT
+keywords = well-known-text, shapely, bokeh, geo-spatial, visualiation
```
Comment (gsornsen) [inline setup.cfg:12]:
```suggestion
keywords = well-known-text, shapely, bokeh, geo-spatial, visualization
```

File: setup.cfg [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+classifiers = 
+    License :: OSI Approved :: MIT License
+    Programming Language :: Python
+    Programming Language :: Python :: 3.7
+    Programming Language :: Python :: 3.8
+    Programming Language :: Python :: 3.9
+    Natural Language :: English
+    Operating System :: OS Independent
```
Comment (gsornsen) [inline setup.cfg:20]: Sounds like I should work on Mac support 😝

File: setup.cfg [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    Development Status :: 5 - Production/Stable
+    Intended Audience :: Developers
+    Topic :: Software Development :: Librarie
```
Comment (gsornsen) [inline setup.cfg:23]:
```suggestion
    Topic :: Software Development :: Libraries
```

To comply with: https://pypi.org/search/?c=Topic+%3A%3A+Software+Development+%3A%3A+Libraries

File: setup.cfg [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+[options]
+include_package_data = True
+package_dir = 
+    =src
```
Comment (gsornsen) [inline setup.cfg:28]: Is this extra `=` necessary?

File: setup.cfg [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    .eggs,
+    __pycache__,
+    tests/,
+    build/,
+    dist/,
+    setup.py,
+    venv/,
+    .github
+ignore = W605, E201, E202, E241
+max-line-length = 120
```
Comment (gsornsen) [inline setup.cfg:54]:
```suggestion
max-line-length = 120

```

**Comment by D0na1d-7rump** [comment]
BLEND IT MIKEY

File: setup.cfg (hunk 7 above)
Comment (misimpso) [reply setup.cfg:28]: Yeah https://setuptools.pypa.io/en/latest/userguide/declarative_config.html?highlight=package_dir#:~:text=described%20above.%0A%0A%5Boptions%5D-,package_dir,-%3D%0A%20%20%20%20%3Dsrc%0Apackages%3Dfind

File: .circleci/config.yml [hunk 9]
```diff
@@ -5,71 +5,12 @@ jobs:
...
-      - run:
-          name: Create .pypirc
-          command: |
-            echo -e "[pypi]" >> ~/.pypirc
-            echo -e "username = __token__" >> ~/.pypirc
-            echo -e "password = $PYPI_TOKEN" >> ~/.pypirc
-      - run:
-          name: Deploy to PyPi
-          command: make upload
+            echo "disable"
```
[the remark this answers is not in the document]
Comment (misimpso) [reply .circleci/config.yml:11]: Yes please.

File: .github/workflows/wktplot.yml [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+  publish:
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/download-artifact@v2
+        with:
+          name: wktplot_dist
+          path: dist
+      - name: Publish package
+        if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
```
[the remark this answers is not in the document]
Comment (misimpso) [reply .github/workflows/wktplot.yml:57]: Let me add an extra condition to make sure the branch is main

File: README.md [hunk 11]
```diff
@@ -1,27 +1,40 @@
...
-
-Plot well-known-text strings and shapely geometries with Bokeh!
+Wrapper around the [Bokeh](https://github.com/bokeh/bokeh) library for plotting well-known-text strings and shapely geometries!
 
 ## Badges
-
-[![CircleCI](https://circleci.com/gh/FuzzFoundation/WKTPlot.svg?style=shield)](https://circleci.com/gh/FuzzFoundation/WKTPlot)
-[![Codacy Badge](https://app.codacy.com/project/badge/Grade/19fe4574645d492e8677c4b06152dd9d)](https://www.codacy.com/gh/FuzzFoundation/WKTPlot/dashboard?utm_source=github.com;utm_content=FuzzFoundation/WKTPlot&amp;utm_campaign=Badge_Grade)
-[![Codacy Badge](https://app.codacy.com/project/badge/Coverage/19fe4574645d492e8677c4b06152dd9d)](https://www.codacy.com/gh/FuzzFoundation/WKTPlot/dashboard?utm_source=github.com&utm_medium=referral&utm_content=FuzzFoundation/WKTPlot&utm_campaign=Badge_Coverage)
+[![codecov](https://codecov.io/gh/FuzzFoundation/WKTPlot/branch/main/graph/badge.svg?token=E1BJVWQLRE)](https://codecov.io/gh/FuzzFoundation/WKTPlot)
```
[the remark this answers is not in the document]
Comment (misimpso) [reply README.md:5]: Badge url is pointing to the main branch by default which doesnt have a codecov report yet.

File: .github/workflows/wktplot.yml [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+  - push
+  - pull_request
+
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+        python-version: ['3.7', '3.8', '3.9']
```
[the remark this answers is not in the document]
Comment (misimpso) [reply .github/workflows/wktplot.yml:13]: Yes that's right.

File: .github/workflows/wktplot.yml [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+        with:
+          python-version: 3.9
+      - name: Create virtual environment
+        run: make init
+      - name: Build module
+        run: make build
+      - uses: actions/upload-artifact@v2
+        with:
+          name: wktplot_dist
+          path: dist/
```
[the remark this answers is not in the document]
Comment (misimpso) [reply .github/workflows/wktplot.yml:46]: I can clamp down on the event triggers now that I know the CI is working. Maybe change the workflow to only run the unittests on pull requests into main, run builds on pushes into main, and run publish on cutting the release (aka tagging the main branch).

File: setup.cfg (hunk 3 above)
Comment (misimpso) [reply setup.cfg:3]: Keep manual bump versioning for now.

File: setup.cfg (hunk 5 above)
Comment (misimpso) [reply setup.cfg:20]: Yes please! Let me add a comment about this in the README

File: requirements.txt (hunk 2 above)
Comment (misimpso) [reply requirements.txt:23]: Let me add this concern to the README.

File: .github/workflows/wktplot.yml (hunk 13 above)
[the remark this answers is not in the document]
Comment (misimpso) [reply .github/workflows/wktplot.yml]: Done!

File: Makefile [hunk 14]
```diff
@@ -1,58 +1,43 @@
...
-	$(VENVPYTHON) -m pip install -r requirements.txt
-
-clean_build:
+ifeq ($(OS),Windows_NT)
+	VENVPYTHON=$(PYTHONVENV)/Scripts/python
+else
+	VENVPYTHON=$(PYTHONVENV)/bin/python
+endif
+
+PHONY: init
```
[the remark this answers is not in the document]
Comment (misimpso) [reply Makefile:9]: I think consolidating them to one line makes each directive look cleaner, however keeping track of them is easier imo when the phony name is with their respective directive logic.

File: .github/workflows/wktplot.yml (hunk 10 above)
[the remark this answers is not in the document]
Comment (misimpso) [reply .github/workflows/wktplot.yml]: Done.

**Review by gsornsen — APPROVED** [review]
LGTM

**Pull request merged**